test/src/e2e_vm_tests/test_programs/should_fail/array_bad_index/test.toml

```
@@ -1,6 +1,115 @@
 category = "fail"
 
-# check: ary[false]
-# nextln: $()Mismatched types.
-# nextln: $()expected: u64
-# nextln: $()found:    bool.
+#check: $()error
+#check: $()array[0u8] = 0;
+#nextln: $()Mismatched types.
+#nextln: $()expected: u64
+#nextln: $()found:    u8.
+#nextln: $()help: Array index must be of type "u64".
+
+#check: $()error
+#check: $()array[0u16] = 0;
+#nextln: $()Mismatched types.
+#nextln: $()expected: u64
+#nextln: $()found:    u16.
+#nextln: $()help: Array index must be of type "u64".
+
+#check: $()error
+#check: $()array[0u32] = 0;
+#nextln: $()Mismatched types.
+#nextln: $()expected: u64
+#nextln: $()found:    u32.
+#nextln: $()help: Array index must be of type "u64".
+
+#check: $()error
+#check: $()array[true] = 0;
```

```suggestion
#nextln: $()array[true] = 0;
```

If you don't require `nextln` here, then you won't catch if `array[0u64] = 0;` results in an error.

---

Unfortunately `#nextln` wouldn't work, because `array[true] = 0;` it is not the immediate next line. But I've did the classical trick, created a bit of a space around `array[0u64] = 0;` and asserted error non-existence with `#not`.